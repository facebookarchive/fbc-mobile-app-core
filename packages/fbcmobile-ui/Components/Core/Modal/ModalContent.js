/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import type {ViewStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet';

import * as React from 'react';
import {StyleSheet, View} from 'react-native';

type Props = {
  +children: React.Node,
  +style?: ViewStyleProp,
};

const ModalContent = ({children, style}: Props) => {
  return <View style={[styles.root, style]}>{children}</View>;
};

const styles = StyleSheet.create({
  root: {
    flexGrow: 1,
    paddingHorizontal: 16,
    paddingBottom: 16,
  },
});

export default ModalContent;
