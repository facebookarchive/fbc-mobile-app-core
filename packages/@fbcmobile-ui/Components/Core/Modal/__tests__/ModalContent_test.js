/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import ModalContent from '../ModalContent';
import React from 'react';
import renderer from 'react-test-renderer';

describe('ModalContent Tests', () => {
  test('ModalContent renders correctly', () => {
    const tree = renderer.create(<ModalContent>{null}</ModalContent>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
