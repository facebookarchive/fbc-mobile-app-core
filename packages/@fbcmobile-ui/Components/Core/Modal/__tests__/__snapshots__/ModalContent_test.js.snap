// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContent Tests ModalContent renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexGrow": 1,
        "paddingBottom": 16,
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
/>
`;
